### ISSUE TITLE
.glb Exporter Fix

### ISSUE THREAD

**Issue by Chadt54** [sergkr/gltflib#109]
Here: https://github.com/sergkr/gltflib/blob/33bb4af035e0a3ede9ece9c84a5a324291582039/gltflib/gltf.py#L507

Since you are removing elements from the list you are currently iterating through, you need to iterate the list in reverse so that the removal of elements don't affect the loop.

Here's the current code:
```python
def _embed_buffer_resources(self):
    if self.model.buffers is None:
        return
    for i, buffer in enumerate(self.model.buffers):
        if buffer.uri is not None:
            resource = self.get_resource(buffer.uri)
            if resource is None:
                raise RuntimeError(f'Missing resource: "{buffer.uri}" (referenced in buffer with index {i})')
            self.embed_resource(resource)
```

Here's a quick fix:
```python
def _embed_buffer_resources(self):
    if self.model.buffers is None:
        return
    for i, buffer in reversed(list(enumerate(self.model.buffers))):
        if buffer.uri is not None:
            resource = self.get_resource(buffer.uri)
            if resource is None:
                raise RuntimeError(f'Missing resource: "{buffer.uri}" (referenced in buffer with index {i})')
            self.embed_resource(resource)
```

**Comment by sergkr** [comment]
Thanks for reporting this. I confirmed there was indeed an issue with embedding buffers due to the collection being modified while iterating. The issue only manifested itself when there are more than 2 buffers that need to be embedded, because embedding involves both removing the original buffer(s) and also adding one for the embedded GLB data. This explains why the issue was masked in existing tests / sample models, since most only have 1 buffer. I fixed the issue and added some tests to cover this scenario.

I kept the original order rather than reversing, so as to preserve the order that the data appears in after it's embedded. The body of the loop doesn't actually use the index (except for error reporting, where we want to report the original index anyway), so I _believe_ that's sufficient - but let me know if there's some scenario I'm missing, or if whatever issue you were originally running into is still not fixed.

FYI, I just released a new version (v1.0.6) which includes this fix.
